### PR TITLE
[WIP]Feature/prepare yaml roundtrip tests

### PR DESCRIPTION
**Pull request by kleinech**
## Improve converter

- Add converter junit5 test cases
- Add derived_from type to ArtifactTypes for round trip
  - either implementation or deployment
- Add WineryPropertiesDefinition while converting from yaml to xml
- Fix NodeTypeImplementation Builder
- Add some comments
- Add some @nullable and @nonnull annotations
- Replace empty exception catches with error logs

- [x] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request. Especially, we require **a single commit**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [x] Ensure that you appear in `NOTICE` at Copyright Holders
- [x] Tests created for changes
- [ ] Documentation updated (if needed)
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

File: org.eclipse.winery.yaml.converter/src/main/java/org/eclipse/winery/yaml/converter/yaml/Y2XConverter.java [hunk 1]
```diff
@@ -203,6 +249,20 @@ private TServiceTemplate convertServiceTemplate(org.eclipse.winery.model.tosca.y
...
+            WinerysPropertiesDefinition properties = new WinerysPropertiesDefinition();
+            PropertyDefinitionKVList propertyDefinitionKVList = new PropertyDefinitionKVList();
+            node.getProperties().forEach((name, property) -> {
+                QName type = TypeConverter.INSTANCE.convert(property.getType());
+                propertyDefinitionKVList.add(new PropertyDefinitionKV(name,
+                    Optional.ofNullable(type.getPrefix())
+                        .map(i -> i.concat(":")).orElse("")
+                        .concat(type.getLocalPart())));
+            });
+            WinerysPropertiesDefinition propertiesDefinition = new WinerysPropertiesDefinition();
```
Comment (lharzenetter) [inline org.eclipse.winery.yaml.converter/src/main/java/org/eclipse/winery/yaml/converter/yaml/Y2XConverter.java]: This is a duplicate. You already introduced a `WineryPropertiesDefinition` variable `properties`  at the top of the `if` which is unused.

File: org.eclipse.winery.yaml.converter/src/main/java/org/eclipse/winery/yaml/converter/Converter.java [hunk 2]
```diff
@@ -114,12 +130,14 @@ public InputStream convertX2Y(DefinitionsChildId id) throws MultiException {
...
     }
 
+    /**
+     * @return A string representation of the YAML after the conversion of the DefinitionsChild to YAML
+     */
     public String convertDefinitionsChildToYaml(DefinitionsChildId id) throws MultiException {
         Path path = Utils.getTmpDir(Paths.get(id.getQName().getLocalPart()));
         convertX2Y(repository.getDefinitions(id), path);
         // convention: single file in root contains the YAML support
         // TODO: Links in the YAML should be changed to real links into Winery
```
[the remark this answers is not in the document]
Comment (kleinech) [reply org.eclipse.winery.yaml.converter/src/main/java/org/eclipse/winery/yaml/converter/Converter.java]: No. This TODO requires more changes:
- X2YConverter optionally create import statements with http request (produce yaml results)
- Y2XConverter support http requests in import statements

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

**Comment by miwurster** [comment]
Closed due to inactivity.

**Pull request closed without merging**